sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/DocumentsImpl.java

```diff
@@ -225,6 +225,19 @@
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Long>> countWithResponseAsync(RequestOptions requestOptions, Context context) {
+        if (this.client.getEndpoint() == null) {
+            return Mono.error(
+                    new IllegalArgumentException(
+                            "Parameter this.client.getEndpoint() is required and cannot be null."));
+        }
+        if (this.client.getIndexName() == null) {
+            return Mono.error(
+                    new IllegalArgumentException(
+                            "Parameter this.client.getIndexName() is required and cannot be null."));
+        }
```

Comment:
This can potentially be validated in the constructor. I have filed an issue for autorest to improve this.
https://github.com/Azure/autorest.java/issues/642